lib/modules/manager/pip-compile/artifacts.ts

```diff
@@ -1,107 +1,32 @@
-
-function getPythonConstraint(
-  config: UpdateArtifactsConfig,
-): string | undefined | null {
-  const { constraints = {} } = config;
-  const { python } = constraints;
-
-  if (python) {
-    logger.debug('Using python constraint from config');
-    return python;
-  }
-
-  return undefined;
-}
-
-function getPipToolsConstraint(config: UpdateArtifactsConfig): string {
-  const { constraints = {} } = config;
-  const { pipTools } = constraints;
-
-  if (is.string(pipTools)) {
-    logger.debug('Using pipTools constraint from config');
-    return pipTools;
-  }
-
-  return '';
-}
-
-const constraintLineRegex = regEx(
-  /^(#.*?\r?\n)+# {4}pip-compile(?<arguments>.*?)\r?\n/,
-);
-const allowedPipArguments = [
-  '--allow-unsafe',
-  '--generate-hashes',
-  '--no-emit-index-url',
-  '--strip-extras',
-];
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+import { extractHeaderCommand, getExecOptions } from './common';
 
 export function constructPipCompileCmd(
   content: string,
-  inputFileName: string,
   outputFileName: string,
 ): string {
-  const headers = constraintLineRegex.exec(content);
-  const args = ['pip-compile'];
-  if (headers?.groups) {
-    logger.debug(`Found pip-compile header: ${headers[0]}`);
-    for (const argument of split(headers.groups.arguments)) {
-      if (allowedPipArguments.includes(argument)) {
-        args.push(argument);
-      } else if (argument.startsWith('--output-file=')) {
-        const file = upath.parse(outputFileName).base;
-        if (argument !== `--output-file=${file}`) {
-          // we don't trust the user-supplied output-file argument; use our value here
-          logger.warn(
-            { argument },
-            'pip-compile was previously executed with an unexpected `--output-file` filename',
-          );
-        }
-        args.push(`--output-file=${file}`);
-      } else if (argument.startsWith('--resolver=')) {
-        const value = extractResolver(argument);
-        if (value) {
-          args.push(`--resolver=${value}`);
-        }
-      } else if (argument.startsWith('--')) {
-        logger.trace(
-          { argument },
-          'pip-compile argument is not (yet) supported',
-        );
-      } else {
-        // ignore position argument (.in file)
-      }
-    }
-  }
-  args.push(upath.parse(inputFileName).base);
-
-  return args.map((argument) => quote(argument)).join(' ');
+  const pipCompileArgs = extractHeaderCommand(content, outputFileName);
+  // TODO(not7cd): sanitize args that require quotes, .map((argument) => quote(argument))
+  return pipCompileArgs.argv.join(' ');
 }
 
 export async function updateArtifacts({
   packageFileName: inputFileName,
   newPackageFileContent: newInputContent,
   config,
 }: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  // TODO(not7cd): must be extracted again or passed from PackageFileContent.lockFiles
   const outputFileName = inputFileName.replace(regEx(/(\.in)?$/), '.txt');
```

Comment:
@rarkins, I see no obvious way to refer to output files. They were used for extraction and passed as `PackageFileContent.lockFiles` but in `updateArtifact(packageFileName, newPackageFileContent, config)` I have no access to that. 
I would also like to point confusing interface name of `PackageFileContent` when `newPackageFileContent` is presumably a string.

Reply:
you need to implement extractAllPackagefiles. then you can return the filename with the extracted deps and extracted args as manager data. this will all be passed to update artifacts later.

Reply:
This is what did here:
https://github.com/renovatebot/renovate/blob/831a7fd2ddc371193e2c234b8d55ead27cedc406/lib/modules/manager/pip-compile/extract.ts#L9-L12

What is the _manager data_?

Reply:
Manager data is an object you can use to pass custom data for a manager from the extract process to the update

Reply:
I found what I needed in `UpdateArtifactsConfig.lockFiles`